lib/puppet/provider/common.rb

```diff
@@ -37,23 +37,29 @@ def data_parse(data = {})
   data
 end
 
-def resource_update(data, resourcetype)
-  current_resource = resourcetype.find_by(@client, unique_id).first
-  return false unless current_resource
-  current_attributes = current_resource.data
-  new_name_validation(data, resourcetype)
-  raw_merged_data = current_attributes.merge(data)
-  updated_data = Hash[raw_merged_data.to_a - current_attributes.to_a]
-  current_resource.update(updated_data) unless updated_data.empty?
-  @property_hash[:data] = current_resource.data
+# Updates resource if it exists and is different from the expected.
+# Returns false if resource does not exist and true if it exists or if it was updated.
+def resource_update
```

Comment:
Because they are already set, accessible, and we're passing in those global variables as the arguments. Since that's the case, it's easier to cut the middle man.